pkg/gardenlet/controller/lease/lease.go

```diff
@@ -61,22 +61,26 @@ func (c *leaseController) Sync(holderIdentity string, ownerReferences ...metav1.
 	}
 	leaseClient := gardenClient.Kubernetes().CoordinationV1().Leases(c.namespace)
 
-	return c.tryUpdateOrCreateLease(ctx, leaseClient, holderIdentity, ownerReferences...)
+	return c.tryCreateOrUpdateLease(ctx, leaseClient, holderIdentity, ownerReferences...)
 }
 
-// tryUpdateOrCreateLease updates or creates the lease if it does not exist.
-func (c *leaseController) tryUpdateOrCreateLease(ctx context.Context, leaseClient coordclientset.LeaseInterface, holderIdentity string, ownerReferences ...metav1.OwnerReference) error {
-	lease, err := leaseClient.Get(ctx, holderIdentity, kubernetes.DefaultGetOptions())
```

Comment:
As this happens quite frequently (every 2s), I propose to instead handle the auth error on get explicitly and do a `create` call in this case instead of doing twice as many calls every 2s now.

Reply:
I would like to prevent such "unauthorized" handling in the client if there is no strong need for it. Another approach could be to remove the `Lease` handler from the RDG and add the corresponding `Lease` vertex (incl. edge) in the `Seed` handler so that the gardenlet is authorized even if it does not yet exist in the system.

Reply:
> Another approach could be to remove the `Lease` handler from the RDG and add the corresponding `Lease` vertex (incl. edge) in the `Seed` handler so that the gardenlet is authorized even if it does not yet exist in the system.

That sounds promising. Let's do that.

Reply:
Generally, that's a nice approach, I think. Adding a vertex/edge for the resources we know that are going to be created by gardenlet, makes sense and simplifies the client-side handling 👍

Reply:
OK, done in f94e6706e403ff32804f050ceb1215a4a6ab256c, please take a look.